pkg/csi/service/wcp/controller_helper.go

```diff
@@ -440,3 +443,88 @@ func getOverlappingNodes(accessibleNodes []string, topologyRequirement *csi.Topo
 	}
 	return overlappingNodes, nil
 }
+
+// canFakeAttach checks if the volume is eligible to be fake attached and returns a bool value
+func canFakeAttach(ctx context.Context, volumeID string, volumeManager cnsvolume.Manager) bool {
+	log := logger.GetLogger(ctx)
+	//Check volume annotations
+	pvcAnn, err := commonco.ContainerOrchestratorUtility.GetVolumeAnnotations(ctx, volumeID)
+	if err != nil {
+		log.Errorf("failed to get volume annotations")
+		return false
```

Comment:
Shouldn't you throw an error? This way the error gets bubbled up to the Pod events.